internal/utils/utils_test.go

```diff
@@ -0,0 +1,43 @@
+// Copyright 2020-2021 Authors of Cilium
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package utils
+
+import "testing"
+
+func TestCheckVersion(t *testing.T) {
+	tests := []struct {
+		name string
+		want bool
+	}{
+		{"v1.9.6", true},
+		{"1.9.6", true},
+		{"1.9", true},
+		{"v1.9", true},
```

Comment:
Interesting. Incidentally, we don't distribute official images with these tags so we should probably prevent these (unless we have some other logic to somehow resolve this to a specific valid tag).